### DISCUSSION TITLE
tuya smartradiator valve HY368

### DISCUSSION THREAD

**Discussion by liviom** [Koenkk/zigbee2mqtt#5445] [Devices]
Dear All

I'm tryning to use z2m with tuya HY368 smart radiator valve, i follow the instruction on the wiki page but when i try to publish with mqttexplorer the message on zigbee2mqtt/FRIENDLY_NAME/set the message for comfort temperature and for eco temperature nothing happen, the temperature value deosn't change

{
      "comfort_temperature": 21
  }

{
      "eco_temperature": 21
  }

for other parameters works fine

how i can solve?

thanks in advance for your help.

**Comment by rafmeeusen** [comment]
Hi, 

I have the same Tuya valve, and I used mosquitto_pub/sub to look what was happening with my mqtt messages. 
Not sure how you concluded that the temp value doesnt change, but you might want to change current_heating_setpoint. 

e.g. 
mosquitto_pub -h localhost -t zigbee2mqtt/0x60a423fffeab4b8a/set/current_heating_setpoint -m "17.5"

eco/comfort temps are used in presets eco and comfort.  (whereas a normal preset is "schedule" when using a weekly schedule)
not sure if these are visible through UI of valve itself...

Hope this helps,
R.

example full message from my device through z2m version 1.17.1: 
{"auto_lock":"AUTO","away_mode":"OFF","away_preset_days":4,"away_preset_temperature":11,"battery_low":false,"boost_time":100,"child_lock":"LOCKED","comfort_temperature":21,"current_heating_setpoint":15,"eco_temperature":12,"force":"normal","holidays":[{"hour":6,"minute":30,"temperature":23},{"hour":9,"minute":0,"temperature":15},{"hour":10,"minute":0,"temperature":15},{"hour":11,"minute":0,"temperature":15},{"hour":12,"minute":0,"temperature":15},{"hour":13,"minute":0,"temperature":15}],"linkquality":15,"local_temperature":19.5,"local_temperature_calibration":0,"max_temperature":29,"min_temperature":6,"position":0,"preset":"schedule","system_mode":"auto","update":{"state":"idle"},"update_available":false,"week":"5+2","window_detection":"ON","window_detection_params":{"minutes":58,"temperature":6},"workdays":[{"hour":6,"minute":0,"temperature":23},{"hour":8,"minute":30,"temperature":15},{"hour":12,"minute":0,"temperature":15},{"hour":13,"minute":0,"temperature":15},{"hour":14,"minute":0,"temperature":15},{"hour":15,"minute":0,"temperature":15}]}
